ibc/modules/src/core/ics03_connection/msgs/conn_open_ack.rs

```diff
@@ -100,8 +100,11 @@ where
 				.map(|height| Height::new(height.revision_number, height.revision_height));
 			if proof_bytes.is_some() && consensus_height.is_some() {
 				Some(
-					ConsensusProof::new(proof_bytes.unwrap(), consensus_height.unwrap())
-						.map_err(Error::invalid_proof)?,
+					ConsensusProof::new(
+						proof_bytes.expect("Contains a value"),
```

Comment:
how are we sure that this doesn't panic?

Reply:
It's contained in an if check before

Reply:
You can use `if let Some(proof_bytes)` so `.expect()` is not needed in the first place, it is more idiomatic in Rust that way.